infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -289,65 +337,62 @@ public void setupTopologyConfigId(long topologyConfigId) {
      * Otherwise, update the base snapshot start timestamp. The update of topologyConfigId just fences off
      * any other metadata updates in other transactions.
      *
+     * @param session unique identifier for LR session
      * @param topologyConfigId current topologyConfigId
-     * @param ts snapshot start timestamp
+     * @param snapshotStartTs snapshot start timestamp
      * @return true, if succeeds
      *         false, otherwise
      */
-    public boolean setBaseSnapshotStart(long topologyConfigId, long ts) {
+    public boolean setBaseSnapshotStart(LogReplicationSession session, long topologyConfigId, long snapshotStartTs) {
+
+        ReplicationMetadata metadata;
+
         try (TxnContext txn = corfuStore.txn(NAMESPACE)) {
-            Map<LogReplicationMetadataType, Long> metadataMap = queryMetadata(txn, LogReplicationMetadataType.TOPOLOGY_CONFIG_ID,
-                    LogReplicationMetadataType.LAST_SNAPSHOT_STARTED);
-            long persistedTopologyConfigID = metadataMap.get(LogReplicationMetadataType.TOPOLOGY_CONFIG_ID);
-            long persistedSnapshotStart = metadataMap.get(LogReplicationMetadataType.LAST_SNAPSHOT_STARTED);
+
+            metadata = queryReplicationMetadata(txn, session);
 
             log.debug("Set snapshotStart topologyConfigId={}, ts={}, persistedTopologyConfigID={}, persistedSnapshotStart={}",
-                    topologyConfigId, ts, persistedTopologyConfigID, persistedSnapshotStart);
+                    topologyConfigId, snapshotStartTs, metadata.getTopologyConfigId(), metadata.getLastSnapshotStarted());
 
             // It means the cluster config has changed, ignore the update operation.
-            if (topologyConfigId != persistedTopologyConfigID) {
-                log.warn("Config differs between sender and receiver, sender[topologyConfigId={}, ts={}]" +
-                                " receiver[persistedTopologyConfigId={}, persistedSnapshotStart={}]", topologyConfigId, ts,
-                        persistedTopologyConfigID, persistedSnapshotStart);
+            if (topologyConfigId != metadata.getTopologyConfigId()) {
+                log.warn("Config differs between source and sink, Source[topologyConfigId={}, ts={}]" +
+                                " Sink[topologyConfigId={}, snapshotStart={}]", topologyConfigId,
+                        snapshotStartTs, metadata.getTopologyConfigId(), metadata.getLastSnapshotStarted());
```

Comment:
since this is an expected path we should do a `txn.commit()` here before returning